go/vt/tabletserver/sysloglogger/sysloglogger.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2016, Google Inc. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package sysloglogger
+
+import (
+	"flag"
+	"fmt"
+	log "github.com/golang/glog"
+	"github.com/youtube/vitess/go/vt/servenv"
+	"github.com/youtube/vitess/go/vt/tabletserver"
+	"log/syslog"
+)
+
+// syslogWriter is an interface that wraps syslog.Writer so it can be faked.
+type syslogWriter interface {
+	Info(string) error
+	Close() error
+}
+
+// writer holds a persistent connection to the syslog daemon
+var writer syslogWriter
+
+var logQueries = flag.Bool("log_queries", false, "Enable query logging to syslog.") // TODO
+
+func init() {
+	servenv.OnRun(func() {
+		if *logQueries {
+			var err error
+			writer, err = syslog.New(syslog.LOG_INFO, "vtquerylogger")
+			if err != nil {
+				log.Error("Query logger is unable to connect to syslog", err)
```

Comment:
For this we usually do:
log.Errorf("query logger is unable to connect to syslog: %v", err)